handlers/handler-publickey-auth/src/main/java/org/apache/servicecomb/authentication/provider/RSAProviderTokenManager.java

```diff
@@ -50,16 +52,16 @@ public boolean valid(String token) {
         LOGGER.error("token is expired");
         return false;
       }
-      if (validatedToken.contains(rsaToken)) {
+      validatedToken = CacheBuilder.newBuilder()
+          .expireAfterAccess(getExpriredTime(), TimeUnit.MILLISECONDS)
+          .build();
+
+      if (validatedToken.asMap().containsKey(rsaToken)) {
         return accessController.isAllowed(MicroserviceInstanceCache.getOrCreate(rsaToken.getServiceId()));
       }
 
-      String sign = rsaToken.getSign();
-      String content = rsaToken.plainToken();
-      String publicKey = getPublicKey(rsaToken.getInstanceId(), rsaToken.getServiceId());
-      boolean verify = RSAUtils.verify(publicKey, sign, content);
-      if (verify && !tokenExprired(rsaToken)) {
-        validatedToken.add(rsaToken);
+      if (isVerify(rsaToken) && !tokenExprired(rsaToken)) {
+        validatedToken.put(rsaToken, true);
```

Comment:
I think this patch does not fix the problem mentiond in this first commit. Each call to "valid", validatedToken is always empty(newly created).